src/app/search/state/search.service.ts

```diff
@@ -271,10 +314,16 @@ export class SearchService {
 
   sortByAlphabet(orderedArray, orderMode): any {
     orderedArray = orderedArray.sort((a, b) => {
+      if (!a.key) {
+        return 1;
+      }
+      if (!b.key) {
+        return -1;
+      }
       if (orderMode) {
-        return a.key > b.key ? 1 : -1;
+        return a.key.toLowerCase() > b.key.toLowerCase() ? 1 : -1;
```

Comment:
I know it was already like this, but since you use it in the other method, I think you should `localeCompare` here as well.